src/qt/rpcconsole.cpp

```diff
@@ -304,9 +304,9 @@ bool RPCConsole::RPCParseCommandLine(std::string &strResult, const std::string &
                             req.strMethod = stack.back()[0];
 #ifdef ENABLE_WALLET
                             // TODO: Move this logic to WalletModel
-                            if (!vpwallets.empty()) {
+                            if (g_wallet_manager->HasWallets()) {
                                 // in Qt, use always the wallet with index 0 when running with multiple wallets
-                                QByteArray encodedName = QUrl::toPercentEncoding(QString::fromStdString(vpwallets[0]->GetName()));
+                                QByteArray encodedName = QUrl::toPercentEncoding(QString::fromStdString(g_wallet_manager->GetWalletAtPos(0)->GetName()));
```

Comment:
Between the above call `HasWallets()` and `GetWalletAtPos()` the wallet can be unloaded. Not possible at the moment, but maybe we should take the opportunity to add a lock to `g_wallet_manager`?

Reply:
Indeed. The locking would be the next step (not only for this case). But its out of scope for this PR.